xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -72,6 +73,18 @@
   private static final String ADS_TYPE_URL_EDS =
       "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
 
+  private final SynchronizationContext syncContext = new SynchronizationContext(
+      new Thread.UncaughtExceptionHandler() {
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
```

Comment:
This `uncaughtException` only logs but does not clean up such as shutting down the channel. Why can't that error handling be done in this PR?

Reply:
I put a TODO there. Current behavior is pretty aggressive, it will panic the JVM. It should be ok, as usually when this happens it is due to some severe bug.

I would need more considerations for how we should handle this more gracefully. Since multiple channels will sharing the same XdsClient instance, it's unclear how uncaught exceptions happening in XdsClient would affect client channels.

Reply:
ok